guides/keras_cv/simsiam-with-kerascv.py

```diff
@@ -0,0 +1,638 @@
+Description: Train a KerasCV model using unlabelled data with SimSiam.
+"""
+
+"""
+## Overview
+
+[TensorFlow similarity](https://github.com/tensorflow/similarity) makes it easy to train
+KerasCV models on unlabelled corpuses of data using contrastive learning algorithms such
+as SimCLR, SimSiam, and Barlow Twins.  In this guide, we will train a KerasCV model
+using the SimSiam implementation from TensorFlow Similarity.
+
+## Background
+
+Self-supervised learning is an approach to pre-training models using unlabeled data. 
+This approach drastically increases accuracy when you have very few labeled examples but
+a lot of unlabelled data.
+The key insight is that you can train a self-supervised model to learn data
+representations by contrasting multiple augmented views of the same example.
+These learned representations capture data invariants, e.g., object translation, color
+jitter, noise, etc. Training a simple linear classifier on top of the frozen
+representations is easier and requires fewer labels because the pre-trained model
+already produces meaningful and generally useful features.
+
+Overall, self-supervised pre-training learns representations which are [more generic and
+robust than other approaches to augmented training and pre-training](https://arxiv.org/abs/2002.05709).
+An overview of the general contrastive learning process is shown below:
+
+![Contrastive overview](https://i.imgur.com/mzaEq3C.png)
+
+In this tutorial, we will use the [SimSiam](https://arxiv.org/abs/2011.10566) algorithm
+for contrastive learning.  As of 2022, SimSiam is the state of the art algorithm for
+contrastive learning; allowing for unprecedented scores on CIFAR-100 and other datasets.
+
+You may need to install:
+
+```bash
+pip -q install tensorflow_similarity
+pip -q install keras-cv
+```
+
+To get started, we will sort out some imports.
+"""
+import resource
+import gc
+import os
+import random
+import time
+import tensorflow_addons as tfa
+import keras_cv
+from pathlib import Path
+import matplotlib.pyplot as plt
+import numpy as np
+from tensorflow.keras import layers
+from tabulate import tabulate
+import tensorflow_similarity as tfsim  # main package
+import tensorflow as tf
+from keras_cv import layers as cv_layers
+
+import tensorflow_datasets as tfds
+
+"""
+Lets sort out some high level config issues and define some constants.
+The resource limit increase is required to load STL-10, `tfsim.utils.tf_cap_memory()`
+prevents TensorFlow from hogging the GPU memory in a cluster, and
+`tfds.disable_progress_bar()` makes tfds less noisy.
+"""
+
+low, high = resource.getrlimit(resource.RLIMIT_NOFILE)
+resource.setrlimit(resource.RLIMIT_NOFILE, (high, high))
+tfsim.utils.tf_cap_memory()  # Avoid GPU memory blow up
+tfds.disable_progress_bar()
+
+BATCH_SIZE = 512
+PRE_TRAIN_EPOCHS = 50
+VAL_STEPS_PER_EPOCH = 20
+WEIGHT_DECAY = 5e-4
+INIT_LR = 3e-2 * int(BATCH_SIZE / 256)
+WARMUP_LR = 0.0
+WARMUP_STEPS = 0
+DIM = 2048
+
+"""
+## Data Loading
```

Comment:
Style nit: typically on keras.io we only capitalize the first word in a title (as well as any word that requires capitalization), not every word